### PR TITLE
init web tests

### PR DESCRIPTION
**Pull request by ransomw1c**
start smoke tests on `pkg/web`.

specifically, the goal here is to land a demo of how to test the `chi` router used to implement the web front.

notice in particular that `pkg/web`'s import of `pkg/core` is mocked out in test in order to test the web server implementation.

### CODE REVIEW

**Comment by ransomw1c** [comment]
the mocks setup here isn't particularly fleshed out.. in the interest of time, in the ignorance of apis.

would be happy to add a little more here for the sake of learning how to use mocks (incl. generated mocks) better.

i suppose this is alright towards the goals of the issue/description?

File: pkg/web/routes.go [hunk 1]
```diff
@@ -121,20 +121,60 @@ func NewServer(params ServerParams) (*Server, error) {
...
+
+func listBundleFilesDefault(repoName string, bundleID string, stores context2.Stores) []model.BundleEntry {
+	bundle := core.NewBundle(core.NewBDescriptor(),
+		core.Repo(repoName),
+		core.ContextStores(stores),
+		core.BundleID(bundleID),
+	)
+	err := core.PopulateFiles(context.Background(), bundle)
+	if err != nil {
+		panic(err)
```
Comment (fredbi) [inline pkg/web/routes.go]: TODO (nice to have): handle error and render error w/ template

File: pkg/web/routes_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	context2 "github.com/oneconcern/datamon/pkg/context"
+	"github.com/oneconcern/datamon/pkg/model"
+
+	//	"github.com/oneconcern/datamon/pkg/web/reverse"
```
Comment (fredbi) [inline pkg/web/routes_test.go]: nit: remove if unused

File: pkg/web/routes_test.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+func setupTests(t *testing.T) http.Handler {
+	coreMocks = newCoreMocks()
+	listRepos = listReposMock
+	listBundles = listBundlesMock
+	listBundleFiles = listBundleFilesMock
+	srv, err := NewServer(ServerParams{})
+	require.NoError(t, err, "create web server instance")
+	//	var srv *Server
```
Comment (fredbi) [inline pkg/web/routes_test.go]: nitpick: remove comment

File: pkg/web/routes.go (hunk 1 above)
Comment (ransomw1c) [reply pkg/web/routes.go]: i've spun off a task here..

File: go.sum [hunk 4]
```diff
@@ -61,6 +65,8 @@ github.com/docker/go-units v0.4.0 h1:3uh0PgVws3nIA0Q+MwDC8yjEPf9zjRfZZWXZYDct3Tw
 github.com/docker/go-units v0.4.0/go.mod h1:fgPhTUdO+D/Jk86RDLlptpiXQzgHJF7gydDDbaIK4Dk=
 github.com/envoyproxy/go-control-plane v0.9.0/go.mod h1:YTl/9mNaCwkRvm6d1a2C3ymFceY/DCBVvsKhRF0iEA4=
 github.com/envoyproxy/protoc-gen-validate v0.1.0/go.mod h1:iSmxcyjqTsJpI2R4NaDN7+kN2VEUnK/pcBlmesArF7c=
+github.com/fredbi/go-github-selfupdate v1.2.0 h1:4wmtpH3u0JGRs7xWf29kdbb6wd2Om6t0DdyD+OWDFek=
```
[the remark this answers is not in the document]
Comment (ransomw1c) [reply go.sum]:
```
% go mod tidy
github.com/oneconcern/datamon/vendor.bkup/google.golang.org/api/googleapi imports
        google.golang.org/api/googleapi/internal/uritemplates: module google.golang.org/api@latest (v0.15.0) found, but does not contain package google.golang.org/api/googleapi/internal/uritemplates
github.com/oneconcern/datamon/vendor.bkup/google.golang.org/api/storage/v1 imports
        google.golang.org/api/gensupport: module google.golang.org/api@latest (v0.15.0) found, but does not contain package google.golang.org/api/gensupport
```

**Comment by ransomw1c** [comment]
> coverage metrics

they're ok, i guess?

https://4394-138348481-gh.circle-artifacts.com/0/coverage-report/coverage_non_cafs.html#file82

the main omission is tests error-handling, as mentioned in the comments.  #377 can include such tests, i suppose?

**Pull request merged**